### ISSUE TITLE
Implement `$lookup` aggregation pipeline stage

### ISSUE THREAD

**Issue by rumyantseva** [FerretDB/FerretDB#1427]
### What should be done?

* https://www.mongodb.com/docs/manual/reference/operator/aggregation/lookup/

**Comment by mrwogu** [comment]
Is there an approximate release date for this solution?

**Comment by AlekSi** [comment]
Not yet. You can change that by voting for the issue description with 👍 and describing your use-case.

**Comment by thesocialdev** [comment]
Hey my application is https://github.com/aletheiafact/aletheia and we have chosen MongoDB while still a open-source application and to move fast with our experiments.

Now, we are tied to its functionality and we don't want to fully re-write everything to a different solution like PostgreSQL.

We want to be able to do a soft migration and possibly through FerretDB, but we make [heavy usage of $lookup pipelines](https://github.com/search?q=repo%3AAletheiaFact%2Faletheia+%24lookup&type=code).

Looking forward to have this supported. How can I know how/when this will been prioritised?

**Comment by AlekSi** [comment]
Soon :)

**Comment by cosmic-flood** [comment]
Hey my application is https://github.com/featbit/featbit and we have chosen MongoDB while still a open-source application and to move fast with our experiments.

Now, we are tied to its functionality and we don't want to fully re-write everything to a different solution like PostgreSQL.

We want to be able to do a soft migration and possibly through FerretDB, but we make [heavy usage of $lookup pipelines](https://github.com/search?q=repo%3AAletheiaFact%2Faletheia+%24lookup&type=code).

Looking forward to have this supported. How can I know how/when this will been prioritised?

**Comment by J44B** [comment]
I am learning web development with javascript and node.js, and since I am using ferretDB as an alternative to the proprietary MongoDB I need this to be working, because it is part of the yourse I'm taking. 
I would guess that is a weak reason to implement this, but... ;-)

**Comment by cosmic-flood** [comment]
> I am learning web development with javascript and node.js, and since I am using ferretDB as an alternative to the proprietary MongoDB I need this to be working, because it is part of the yourse I'm taking. I would guess that is a weak reason to implement this, but... ;-)

One year after...  is this project still alive?

**Comment by AlekSi** [comment]
Yes, it is alive. We are working on the next major version privately for now. It is going to be open-sourced very soon. It will support `$lookup` among many other things.

**Comment by AlekSi** [comment]
This is now supported in [v2.0.0-rc.1](https://github.com/FerretDB/FerretDB/releases/tag/v2.0.0-rc.1)!
